### PR TITLE
Add Mobile Navbar

### LINKED ISSUE #27: Add mobile navbar
- We need the navbar on the mobile view
- This involves responsive designs (see individual recipe page for an example of how to hide/show certain elements based on viewport size)
- Consider a Chakra `Drawer` (https://chakra-ui.com/docs/overlay/drawer) for the side nav menu
- See Figma for relevant designs:
  - https://www.figma.com/file/6SnGmwXgJwpRojgPWXKkYn/Health-Families-Healthy-Futures-(HF)-%2F-Spr21?node-id=1%3A5
  - https://www.figma.com/file/6SnGmwXgJwpRojgPWXKkYn/Health-Families-Healthy-Futures-(HF)-%2F-Spr21?node-id=410%3A0

### PR DESCRIPTION
**Pull request by ibaniya98**
- Refactored `Navigation` component to display navigation links
- Used `Drawer` component to show side navbar in mobile view.

### CODE REVIEW

File: src/components/Navigation.tsx [hunk 1]
```diff
@@ -1,109 +1,153 @@
...
-    <Flex
-      h="55px"
-      wrap="nowrap"
-      mx="auto"
-      flexDir="row"
-      w="full"
-      justifyContent="space-evenly"
-      alignItems="center"
+    <Link
+      _hover={{
```
Comment (migueldlr) [inline src/components/Navigation.tsx]: Could these styles (and the styles for `MobileNavLink` be migrated to `theme.ts`? The components seem like they're just wrappers for `Link`

File: src/components/Navigation.tsx [hunk 2]
```diff
@@ -1,109 +1,153 @@
...
-        fontWeight="850"
-        fontSize="16px"
-        as={GatsbyLink}
-        to="/recipes/"
-      >
-        RECIPES
-      </Link>
+function Navigation({ data }: NavigationProps): JSX.Element {
+  const navigationItems: { [link: string]: string } = {
+    '/': 'HOME',
```
Comment (migueldlr) [inline src/components/Navigation.tsx]: These can just be regular case - we can add `text-transform: uppercase` on the actual component to stylize (https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform)

File: src/components/Navigation.tsx [hunk 3]
```diff
@@ -1,109 +1,153 @@
...
+                    );
+                  })}
+                </Flex>
+              </DrawerBody>
+            </DrawerContent>
+          </DrawerOverlay>
+        </Drawer>
+
+        <HamburgerIcon h={6} w={6} color="gray.700" onClick={onOpen} />
+        <Box w="178px" display={['block', null, 'none']} margin="auto">
```
Comment (migueldlr) [inline src/components/Navigation.tsx]: I don't love the idea of having the image in `Navigation` in addition to `Header`, but c'est la vie

Comment (ibaniya98) [reply src/components/Navigation.tsx]: Updated the layout to move the HFHF Image to the header on mobile view.

File: src/components/Navigation.tsx (hunk 1 above)
Comment (ibaniya98) [reply src/components/Navigation.tsx]: Created a styling for `NavLink` in `theme.ts`. It currently uses `useStyleConfig` to apply the styles in our custom component.

Using variant in `Link` by itself was not helpful considering different styles for desktop/mobile view and we cannot have multiple variants.

**Review by migueldlr — APPROVED** [review]

**Pull request merged**